From ff928defab4b8d4522a93cdd8c98116137744a43 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Sun, 23 Jul 2017 08:57:57 -0500
Subject: [PATCH] Fallback to using ruamel.yaml in conda skeleton

Use the namespaced ruamel.yaml import in conda skeleton pypi if
importing ruamel_yaml fails.  If both fail to import raise a helpful
error message.

Import logic adapted from common/serialize.py file in conda source code.
---
 conda_build/skeletons/pypi.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/conda_build/skeletons/pypi.py b/conda_build/skeletons/pypi.py
index a1ac64703..daee9cdef 100644
--- a/conda_build/skeletons/pypi.py
+++ b/conda_build/skeletons/pypi.py
@@ -21,7 +21,14 @@
 from requests.packages.urllib3.util.url import parse_url
 from six.moves.urllib.parse import urljoin
 import yaml
-import ruamel_yaml
+try:
+    import ruamel_yaml
+except ImportError:
+    try:
+        import ruamel.yaml as ruamel_yaml
+    except ImportError:
+        raise ImportError("No ruamel_yaml library available.\n"
+                          "To proceed, conda install ruamel_yaml")
 
 from conda_build.conda_interface import spec_from_line
 from conda_build.conda_interface import input, configparser, StringIO, string_types, PY3
